backend/python_init_script/main.py

```diff
@@ -2,41 +2,186 @@
 # SPDX-FileCopyrightText: 2021 Felix Steinkohl <steinkohl@campus.tu-berlin.de>
 
 import os
+from dataclasses import dataclass
+import pandas as pd
 from appwrite.client import Client
 from appwrite.services.users import Users
 from appwrite.services.teams import Teams
+from appwrite.exception import AppwriteException
+
+
+@dataclass
+class UserData:
+    """
+    Dataclass for specifying user data
+    """
+
+    name: str
+    email: str
+    team: str
+    roles: [str]
+    init_pw: str
+
+
+def load_user_data(file: str) -> [UserData]:
+    """
+    Reads the given file and parses it into UserData objects.
+
+    :param file: path of file containing the user data that shall be loaded
+    :type file: str
+    :return: iterable of the loaded UserData objects
+    :rtype: [UserData]
+    """
+    ending = str(file).lower().split(".")[-1]
+    user_df = pd.DataFrame()
+    if ending == "csv":
+        user_df = pd.read_csv(file, dtype=str)
+        # if there are less than 5 columns -> there might be ';' used as separators
+        if user_df.shape[1] < 5:
+            user_df = pd.read_csv(file, sep=";", dtype=str)
+    elif ending in ["xlsx", "xls"]:
+        user_df = pd.read_excel(file, dtype=str)
+    else:
+        print("Only csv and xlsx format are supported!")
+
+    for row_id, row in user_df.iterrows():
+        # if cell has no value in xlsx/csv it will result in a nan value
+        # -> replace with empty string
```

Comment:
Redundant comments here. From a software-engineering point, these are details cluttered into a bigger function which would better fit into their own functions for better level-of-detail. Functions are very useful for readability so writing small functions is totally fine and even recommended by experienced software engineers.

```
_roles = splitRoles(row)
_team = getTeam(row)
```

Reply:
fixed.
Added get_roles() and more generic _replace_nan() for the rest.
btw, python always uses lower_case_underscore notation for methods, only classes are CamelCase (PEP8)